src/app/gui/queryresults/queryresultsservice.js

```diff
@@ -1373,44 +1372,45 @@ proto.showRelationsChart = function(ids=[], layer, feature, action, index, conta
   } else this.hideChart(container)
 };
 
-proto.printAtlas = function(layer, feature){
-  let {id:layerId, features} = layer;
+proto.printAtlas = function(layer, feature) {
+  let {features} = layer;
+  features = feature ? [feature] : features;
```

Comment:
This assignment can mask unexpected side effects.

In this case I recommend making it more explicit what the desired behavior is:


```suggestion
const features   = feature ? [feature] : layer.features;
```

```suggestion
layer.features   = feature ? [feature] : layer.features;
```

Reply:
@volterra79 please see you how to resolve this "ambiguity", the rest it seems ok to me.